static/app/views/replays/detail/trace/replayTransactionContext.tsx

```diff
@@ -0,0 +1,212 @@
+  state: INITIAL_STATE,
+});
+
+function ReplayTransactionContext({children, replayRecord}: Options) {
+  const api = useApi();
+  const location = useLocation();
+  const organization = useOrganization();
+
+  const [state, setState] = useState<State>(INITIAL_STATE);
+
+  const orgSlug = organization.slug;
+
+  const listEventView = useMemo(() => {
+    if (!replayRecord) {
+      return null;
+    }
+    const replayId = replayRecord?.id;
+    const projectId = replayRecord?.project_id;
+    const start = getUtcDateString(replayRecord?.started_at.getTime());
+    const end = getUtcDateString(replayRecord?.finished_at.getTime());
+
+    return EventView.fromSavedQuery({
+      id: undefined,
+      name: `Traces in replay ${replayId}`,
+      fields: ['trace', 'count(trace)', 'min(timestamp)'],
+      orderby: 'min_timestamp',
+      query: `replayId:${replayId}`,
+      projects: [Number(projectId)],
+      version: 2,
+      start,
+      end,
+    });
+  }, [replayRecord]);
+
+  const tracePayload = useMemo(() => {
+    const start = getUtcDateString(replayRecord?.started_at.getTime());
+    const end = getUtcDateString(replayRecord?.finished_at.getTime());
+
+    const traceEventView = makeEventView({start, end});
+    return getTraceRequestPayload({eventView: traceEventView, location});
+  }, [replayRecord, location]);
+
+  const fetchTraceIdData = useCallback(
+    async (traceIds: string[]) => {
+      setState(prev => ({
+        ...prev,
+        detailsRequests: prev.detailsRequests + traceIds.length,
+      }));
+
+      await Promise.allSettled(
+        traceIds.map(traceId => {
+          const promise = doDiscoverQuery(
+            api,
+            `/organizations/${orgSlug}/events-trace/${traceId}/`,
+            tracePayload
+          );
+          promise.then(([trace, , _traceResp]) => {
+            setState(prev => ({
+              ...prev,
+              detailsResponses: prev.detailsResponses + 1,
+              traces: sortBy(
+                (prev.traces || []).concat(trace as TraceFullDetailed),
+                'start_timestamp'
+              ),
+            }));
+          });
+          return promise;
+        })
+      );
+
+      setState(prev => ({
+        ...prev,
+        detailsComplete: prev.detailsRequests === prev.detailsResponses,
+      }));
+    },
+    [api, orgSlug, tracePayload]
+  );
+
+  const fetchTransactionData = useCallback(async () => {
+    if (state.traces || !listEventView) {
+      return;
+    }
+    const start = getUtcDateString(replayRecord?.started_at.getTime());
+    const end = getUtcDateString(replayRecord?.finished_at.getTime());
+
+    setState({
+      error: null,
+      isFetching: true,
+      indexComplete: false,
+      detailsRequests: 0,
+      detailsResponses: 0,
+      detailsComplete: false,
+      traces: [],
+    });
+
+    let cursor = {
+      cursor: '0:0:0',
+      results: true,
+      href: '',
+    } as ParsedHeader;
+    while (cursor.results) {
```

Comment:
A bit concerned about the # of requests for enterprise orgs (could be 400+ requests)

Reply:
I agree.
It's going to break right away, and if we wait for _after_ ingest adds a bias for replays then it'll break more often and we will be catching up trying to fix it in Q3. 

IMO we should be re-building this tab almost from scratch in Q2. It needs to be searchable and filterable so you can grok through hundreds of traces, and load much less data in the process. That's going to need new api calls probably, etc.